### ISSUE TITLE
用huggingface代码直接进行BART large fineturning出现繁体字

### ISSUE THREAD

**Issue by yht4work** [fastnlp/CPT#33]
以下为训练集的数据，训练了1000epoch，可以看到不仅`预算`变成了`預算`（繁简），而且`A=SM`变成了`a=sm`（大小写），也就是连训练集都没有拟合，训练过程loss是接近于0的

生成: 题目：《sm公司全面预算管理问题研究》，句式：a，其中a=sm公园公司的全面預算管辖问题探究
label: 题目：《SM 公司全面预算管理问题研究》，句式：A，其中A=SM 公司全面预算管理问题研究

想问下可能的原因

**Comment by choosewhatulike** [comment]
可能是数据处理的问题，Tokenizer和Model都不会对简繁体进行转换，而英文会统一转成小写。

**Comment by yht4work** [comment]
```
def tokenize_function(examples):
    return tokenizer(examples["text"], max_length=512, padding="max_length", truncation=True)

raw_datasets = load_dataset(extension, data_files=data_files)
tokenized_datasets = raw_datasets.map(tokenize_function, batched=True)
```
jsonl数据直接通过datasets加载后直接转id，没有进行其他预处理。想问下有没有其他可能性

**Comment by choosewhatulike** [comment]
可能是datasets内部的其它部分？可以打印一些中间结果看看

**Comment by yht4work** [comment]
训练代码
```python
import os
from datasets import load_dataset
from transformers import BertTokenizer, BartForConditionalGeneration, Seq2SeqTrainingArguments, Seq2SeqTrainer, DataCollatorForSeq2Seq
# =========定义常量=========
path = "/yuhang/draft/models/NLP/bart-large-chinese"
text_column = "input_text"
summary_column = "output_text"
max_source_length = 384
max_target_length = 384
padding = "max_length"
ignore_pad_token_for_loss = True
num_train_epochs = 100
os.environ["CUDA_VISIBLE_DEVICES"] = "7"
data_files = {}
data_files["train"] = "/paper/title/train/train.json"
data_files["validation"] = "/paper/title/train/test.json"
model_output_dir = "models/trained/bart-large-chinese_ft_2022.3.9.8.38"
os.system("rm -rf %s" % model_output_dir)
# ========初始化模型========
tokenizer = BertTokenizer.from_pretrained(pretrained_model_name_or_path=path)
model = BartForConditionalGeneration.from_pretrained(
    pretrained_model_name_or_path=path)
data_collator = DataCollatorForSeq2Seq(
    tokenizer,
    model=model,
    pad_to_multiple_of=8
)
# =========text转token id=========
def preprocess_function(examples):
    inputs = examples[text_column]
    targets = examples[summary_column]
    inputs = [inp for inp in inputs]
    model_inputs = tokenizer(
        inputs, max_length=max_source_length, padding=padding, truncation=True)
    # Setup the tokenizer for targets
    # with tokenizer.as_target_tokenizer():
    labels = tokenizer(targets, max_length=max_target_length,
                       padding=padding, truncation=True)
    # If we are padding here, replace all tokenizer.pad_token_id in the labels by -100 when we want to ignore
    # padding in the loss.
    if padding == "max_length" and ignore_pad_token_for_loss:
        labels["input_ids"] = [[(l if l != tokenizer.pad_token_id else -100)
                                for l in label] for label in labels["input_ids"]]
    model_inputs["labels"] = labels["input_ids"]
    return model_inputs


# =========加载数据=========
extension = data_files["train"].split(".")[-1]
raw_datasets = load_dataset(extension, data_files=data_files)
tokenized_datasets = raw_datasets.map(preprocess_function, batched=True)
# ==========训练==========
training_args = Seq2SeqTrainingArguments(
    output_dir=model_output_dir,
    evaluation_strategy="epoch",
    learning_rate=1e-5,
    per_device_train_batch_size=8,
    per_device_eval_batch_size=8,
    weight_decay=0.01,
    save_total_limit=3,
    num_train_epochs=num_train_epochs,
    fp16=True,
)

trainer = Seq2SeqTrainer(
    model=model,
    args=training_args,
    train_dataset=tokenized_datasets["train"],
    eval_dataset=tokenized_datasets["validation"],
    tokenizer=tokenizer,
    data_collator=data_collator,
)
# ==========保存模型和tokenizer==========
trainer.train()
trainer.save_model()  # Saves the tokenizer too

```

**Comment by yht4work** [comment]
> 可能是datasets内部的其它部分？可以打印一些中间结果看看

确实，print算是通用的debug方式了，hhh

**Comment by choosewhatulike** [comment]
> 以下为训练集的数据，训练了1000epoch，可以看到不仅预算变成了預算（繁简），而且A=SM变成了a=sm（大小写），也就是连训练集都没有拟合，训练过程loss是接近于0的

loss接近于0应该有拟合训练集？可能只是全变成繁体了

**Comment by TheHonestBob** [comment]
> 以下为训练集的数据，训练了1000epoch，可以看到不仅`预算`变成了`預算`（繁简），而且`A=SM`变成了`a=sm`（大小写），也就是连训练集都没有拟合，训练过程loss是接近于0的
> 
> 生成: 题目：《sm公司全面预算管理问题研究》，句式：a，其中a=sm公园公司的全面預算管辖问题探究 label: 题目：《SM 公司全面预算管理问题研究》，句式：A，其中A=SM 公司全面预算管理问题研究
> 
> 想问下可能的原因
请问最后找到原因了吗，我也是用于生成，生成的文本里面有不少繁体字，找不到原因目前
